theme_report/generate_theme_report.py

```diff
@@ -40,78 +34,86 @@ def parse_subaccount_theme(account, driver, API_URL, report_df):
+                                    logging.info(f'{account.id};  {account.name}; {theme["name"]}')
+                                    report_df = pd.concat([report_df, theme_df])
     return report_df
 
 
-def loop_subaccount(account, driver, API_URL, report_df):
+def loop_subaccount(account, page, API_URL, report_df):
     """
     iterating through all subaccounts recursively
     """
-    print(f"""current account {account} {account.id}""")
+    logging.info(f"""current account {account} {account.id}""")
     subaccounts = account.get_subaccounts()
     for sa in subaccounts:
-        report_df = loop_subaccount(sa, driver, API_URL, report_df)
-        report_df = parse_subaccount_theme(sa, driver, API_URL, report_df)
-        print(f"{report_df.shape}")
+        # recursively loop subaccount
+        report_df = loop_subaccount(sa, page, API_URL, report_df)
 
-    return report_df
+        # get the subaccount theme info
+        report_df = parse_subaccount_theme(sa, page, API_URL, report_df)
 
+    return report_df
 
-def main():
+def get_session_url(canvas):
+    response = canvas._Canvas__requester.request(
+        'GET', 
+        _url = canvas._Canvas__requester.original_url + "/login/session_token"
+    )
+    # If response.text is json the return the value session_url
+    if response.text:
+        session_url = response.json().get('session_url')
+        return session_url
+    else:
+        # Log an error that it couldn't get the response_url
+        logging.error("Error getting session_url")
+        return None
+
+def run(playwright: Playwright) -> None:
     # Set up ENV
     CONFIG_PATH = os.path.join(os.path.dirname(
         os.path.abspath(__file__)), './env.json')
     try:
         with open(CONFIG_PATH) as env_file:
             ENV = json.load(env_file)
     except FileNotFoundError:
-        logger.error(
+        logging.error(
             f'Configuration file could not be found; please add file "{CONFIG_PATH}".')
         ENV = dict()
 
     # get the API parameters from ENV
     API_URL = ENV["API_URL"]
     API_KEY = ENV["API_KEY"]
 
-    # used for login page
-    CANVAS_USER = ENV["CANVAS_USER"]
-    CANVAS_PASSWORD = ENV["CANVAS_PASSWORD"]
-
-    # Initialize the driver
-    driver = webdriver.Chrome()
-
-    # Open provided link in a browser window using the driver
-    driver.get(API_URL + "/login/saml")
-    driver.execute_script(
-        f"document.getElementById('login').value = '{CANVAS_USER}';")
-    driver.execute_script(
-        f"document.getElementById('password').value = '{CANVAS_PASSWORD}';")
-    loginSubmit = driver.find_element_by_id('loginSubmit')
-    loginSubmit.click()
-
-    # create an empty dataframe
-    report_df = pd.DataFrame(
-        columns=('account_id', "account_name", "current_theme_name"))
-
     # Initialize a new Canvas object
     canvas = Canvas(API_URL, API_KEY)
+    session_url = get_session_url(canvas)
+
+    browser = playwright.chromium.launch(headless=False)
```

Comment:
I might make this headless value configurable in the json but doesn't have to happen here.